devices/Paexo/src/Paexo.cpp

```diff
@@ -187,6 +236,29 @@ Paexo::Paexo()
 // Destructor
 Paexo::~Paexo() = default;
 
+class Paexo::PaexoImpl::PaexoMotorControlPort : public YarpBufferedPort
+{
+public:
+
+    std::string portName;
+    ElementPtr<Paexo::PaexoImpl::PaexoMotorActuator> paexoMotorActuator = nullptr;
+
+    void onRead(yarp::os::Bottle& motorCommand) override;
+
+    // Constructor
+    PaexoMotorControlPort(const std::string& name, ElementPtr<Paexo::PaexoImpl::PaexoMotorActuator>& actuator)
+    {
+        portName = name;
+        // Set the actutor
+        if (actuator == nullptr)
+        {
+            yError() << LogPrefix << "Actuator passing error for motor control port " << portName;
+        }
```

Comment:
This is not the best pattern, what would happen in the `onRead` if this occurs? If you don't want to add a new `bool initialize()` methods so that a `false` could be propagated, maybe throwing a `std::runtime_error` is the best solution.

Reply:
> This is not the best pattern, what would happen in the onRead if this occurs? 

There is an `assert` in `onRead` to check this case. Here just added an error log message.

Reply:
Asserts are not compiled in any build type that is not Debug, relying on them is not a good pattern. With any other build types, you'll likely get a segfault. Without changing too much the logic, maybe adding a:

```cpp
#include <stdexcept>
/// [...]
if ()
{
  throw std::runtime_error("<message>")
}
```

is safer. This applies also to other similar cases in the future. As a rule of thumbs, it's better an exception with a clear message rather than a segfault.

Reply:
Done in https://github.com/robotology/wearables/pull/124/commits/034fbeab4877befdbfdd4e5034ab43a95cbb715f